lib/backend/local/local.go

```diff
@@ -0,0 +1,331 @@
+	return &LocalBackend{
+		file: file,
+	}
+}
+
+func (lb *LocalBackend) GetImageInfo(dockerURL string) ([]string, *types.ParsedDockerURL, error) {
+	parsedDockerURL := common.ParseDockerURL(dockerURL)
+
+	appImageID, parsedDockerURL, err := getImageID(lb.file, parsedDockerURL)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error getting ImageID: %v", err)
+	}
+
+	ancestry, err := getAncestry(lb.file, appImageID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error getting ancestry: %v", err)
+	}
+
+	return ancestry, parsedDockerURL, nil
+}
+
+func (lb *LocalBackend) BuildACI(layerID string, dockerURL *types.ParsedDockerURL, outputDir string, curPwl []string) (string, *schema.ImageManifest, error) {
+	tmpDir, err := ioutil.TempDir("", "docker2aci-")
+	if err != nil {
+		return "", nil, fmt.Errorf("error creating dir: %v", err)
+	}
+
+	j, err := getJson(lb.file, layerID)
+	if err != nil {
+		return "", nil, fmt.Errorf("error getting image json: %v", err)
+	}
+
+	layerData := types.DockerImageData{}
+	if err := json.Unmarshal(j, &layerData); err != nil {
+		return "", nil, fmt.Errorf("error unmarshaling layer data: %v", err)
+	}
+
+	tmpLayerPath := path.Join(tmpDir, layerID)
+	tmpLayerPath += ".tar"
+
+	layerFile, err := extractEmbeddedLayer(lb.file, layerID, tmpLayerPath)
+	if err != nil {
+		return "", nil, fmt.Errorf("error getting layer from file: %v", err)
+	}
+	defer layerFile.Close()
+
+	util.Debug("Generating layer ACI...")
+	aciPath, manifest, err := common.GenerateACI(layerData, dockerURL, outputDir, layerFile, curPwl)
+	if err != nil {
+		return "", nil, fmt.Errorf("error generating ACI: %v", err)
+	}
+
+	return aciPath, manifest, nil
+}
+
+func getImageID(file *os.File, dockerURL *types.ParsedDockerURL) (string, *types.ParsedDockerURL, error) {
+	type tags map[string]string
+	type apps map[string]tags
+
+	_, err := file.Seek(0, 0)
+	if err != nil {
+		return "", nil, fmt.Errorf("error seeking file: %v", err)
+	}
+
+	var imageID string
+	var appName string
+	reposWalker := func(t *tarball.TarFile) error {
+		cleanName := filepath.Clean(t.Name())
+
+		if cleanName == "repositories" {
+			repob, err := ioutil.ReadAll(t.TarStream)
+			if err != nil {
+				return fmt.Errorf("error reading repositories file: %v", err)
+			}
+
+			var repositories apps
+			if err := json.Unmarshal(repob, &repositories); err != nil {
+				return fmt.Errorf("error Unmarshaling repositories file")
+			}
+
+			if dockerURL == nil {
+				if len(repositories) > 1 {
+					var appNames []string
+					for key, _ := range repositories {
+						appNames = append(appNames, key)
+					}
+					return fmt.Errorf("several images found, use option --image with one of:\n\n%s", strings.Join(appNames, "\n"))
+				} else {
+					for key, _ := range repositories {
+						appName = key
+					}
+				}
+			} else {
+				appName = dockerURL.ImageName
+			}
+
+			tag := "latest"
+			if dockerURL != nil {
+				tag = dockerURL.Tag
+			}
+
+			if _, ok := repositories[appName]; !ok {
```

Comment:
be optimistic and assign the value to something so you don't have to keep dereferencing it again below